### ISSUE TITLE
View field "raw" value wrongly returns node ID

### ISSUE THREAD

**Issue by docwilmot** [backdrop/backdrop-issues#5850]
## Description of the bug

I'm testing a views template row output with in a theme. Its a View of fields. Debugging fields raw output is returning the NID instead of the raw field content. So `dpm($fields['body']->raw` returns `2` for the "About" node. instead of the body content.

## Steps To Reproduce

1. create a View of nodes. Under format, be sure to choose `fields` instead of `content`
2. add the body field, or any other fields (the node title field doesnt seem to have this problem)
3. you need to override the row template (maybe other templates have same issue, but havent checked). Under `Theme: information` will be the available `row style` templates (such as  `views-view-fields.tpl.php`, `views-view-fields--default.tpl.php`, etc). Create a file in your theme `templates` folder with one of these template names (I tested with the `views-view-fields.tpl.php` template).
4. check the output of fields' `raw` value. In that template I added a `dpm($fields['body']->raw)`

## Actual behavior

Note that this returns the node ID

## Expected behavior

`$fields['ANY_FIELD']->raw` should return the raw value of that field, not the node ID. 

## Additional information

Add any other information that could help, such as:
- Backdrop CMS version: 1.23

**Comment by docwilmot** [comment]
Confirmed by @argiepiano. See conversation on Zulip https://backdrop.zulipchat.com/#narrow/stream/218635-Backdrop/topic/views.20fields.20raw.20output/near/307359281

**Comment by indigoxela** [comment]
@docwilmot can you please also provide the raw query (you can turn the display on on /admin/structure/views/settings).

In a very simple example view I get:

```
SELECT node.`title` AS `node_title`, node.`nid` AS `nid`, 'node' AS `field_data_body_node_entity_type`
FROM 
{node} node
WHERE (( (`node`.`status` = '1') AND (`node`.`type` IN  ('post')) ))
```

BTW I can also confirm, that the "raw" property is a bit unexpected:

```
      debug($fields['body']->content); // content from database
      debug($fields['body']->raw); // node ID
```
But with views it's always a bit hard to determine, if it's a bug or just odd behavior or misleading variable naming. :wink:

**Comment by indigoxela** [comment]
Here's some [info by merlinofchaos](https://www.drupal.org/project/views/issues/1160706#comment-4916402).

That was something, I already suspected (a bit), when looking at the query. Fields (also body) are in different database tables.

The "raw" value here probably isn't supposed to be the "unchanged value from table field_data_body", as that's not part of the main query, "raw" just points to the entity ID. Note: "raw" from the main db query perspective - if I got things right.

**Comment by argiepiano** [comment]
Yes, that clarifies it. So perhaps we could modify the documentation of `views-view-fields.tpl.php`. It currently states:

` *   - $field->raw: The raw data for the field, if it exists. This is NOT output safe.`

(I guess the "if it exists" is probably enough to give you pause about using this value, but still, it may be appropriate to say something like "the raw data or the entity ID...".  I'm pretty sure that in the case of table columns, like `title`, `raw` does contain the raw content of the column. Thus, the confusion.
